### PR TITLE
Add integrity demo with sample production event

### PR DESCRIPTION
**Pull request by honourfish**
## Testing

local testing using `cd integrity && go run .` yields:

```
Event included on merkle log: true
```
re: AB#9491

### CODE REVIEW

File: integrity/go.mod [hunk 1]
```diff
@@ -2,10 +2,77 @@ module github.com/datatrails/go-datatrails-demos/integrity
 
 go 1.22.2
 
+// TODO remove replace when it goes public, and depend on the released version of logverification
+replace github.com/datatrails/go-datatrails-demos/logverification => ../logverification
```
Comment (honourfish) [inline integrity/go.mod:6]: Will update when we get a new release out.

File: integrity/go.mod [hunk 2]
```diff
@@ -2,10 +2,77 @@ module github.com/datatrails/go-datatrails-demos/integrity
...
+
+require (
+	github.com/datatrails/go-datatrails-common v0.15.4-0.20240524103926-4968ec778880
```
Comment (honourfish) [inline integrity/go.mod:9]: will update when we get a new release out.

File: README.md [hunk 3]
```diff
@@ -1 +1,37 @@
...
+The integrity demo will verify the integrity of a datatrails event.
+
+This is achieved by creating an inclusion proof for that event and
+then verifying the inclusion proof against the merkle log.
+
+If the inclusion proof is verified successfully, we can say that the
+datatrails event is included on the merkle log.
+
+### Dockerised Demo
+To run the integrity demo without local golang install:
```
Comment (henry739) [inline README.md:14]: Might want to generalise the wording here

Comment (honourfish) [reply README.md]: have tried to generalise it :)

File: integrity/main.go [hunk 4]
```diff
@@ -1,5 +1,112 @@
...
+	`
+
+	publicTenantID = "tenant/6ea5cd00-c711-3649-6914-7b125928bbb4"
+
+	// merklelog reader configuration
+	container = "merklelogs"
+	url       = "https://app.datatrails.ai/verifiabledata"
+)
+
+// IntegrityDemoDemo of a public datatrails event
```
[the remark this answers is not in the document]
Comment (honourfish) [reply integrity/main.go]: removed the extra Demo!

**Pull request merged**